alacritty/src/display.rs

```diff
@@ -216,19 +214,14 @@ impl Display {
         let (glyph_cache, cell_width, cell_height) =
             Self::new_glyph_cache(dpr, &mut renderer, config)?;
 
-        let padding = config.ui_config.window.padding;
-        let mut padding_x = f32::from(padding.x) * dpr as f32;
-        let mut padding_y = f32::from(padding.y) * dpr as f32;
+        let (mut padding_x, mut padding_y) = scale_padding(config, dpr);
 
-        if let Some((width, height)) =
-            GlyphCache::calculate_dimensions(&config.ui_config.window, dpr, cell_width, cell_height)
-        {
-            let PhysicalSize { width: w, height: h } = window.inner_size();
-            if w == width && h == height {
-                info!("Estimated DPR correctly, skipping resize");
-            } else {
-                window.set_inner_size(PhysicalSize::new(width, height));
-            }
+        if (estimated_dpr - dpr).abs() < f64::EPSILON {
```

Comment:
That seems wrong, since you're skipping dynamic padding whenever dimensions are `None` (e.g. Fullscreen startup, etc).

Reply:
There is no with_fullscreen, the only window size option at startup is either dimensions or maximized. When maximized is set, we do not have any dimensions. So the only scenario where dimensions is Some, is when the window is a normal window that has fixed dimensions and we don't need dynamic padding there.